src/vs/workbench/contrib/chat/browser/actions/chatClearActions.ts

```diff
@@ -167,17 +168,54 @@ export function registerNewChatActions() {
 				icon: Codicon.redo,
 				precondition: ContextKeyExpr.and(ChatContextKeys.chatEditingCanRedo, ChatContextKeys.enabled, ChatContextKeys.editingParticipantRegistered),
 				f1: true,
+				menu: [
+					{
+						id: MenuId.ViewTitle,
+						when: ContextKeyExpr.equals('view', ChatViewId),
+						group: 'navigation',
+						order: -2,
+						isHiddenByDefault: true
+					}
+				]
+			});
+		}
+
+		async runEditingSessionAction(accessor: ServicesAccessor, editingSession: IChatEditingSession) {
+			await editingSession.redoInteraction();
+		}
+	});
+
+	registerAction2(class RedoChatEditInteractionAction2 extends EditingSessionAction {
+		constructor() {
+			super({
+				id: 'workbench.action.chat.redoEdit2',
+				title: localize2('chat.redoEdit.label2', "Redo Last Request"),
+				category: CHAT_CATEGORY,
+				precondition: ContextKeyExpr.and(ChatContextKeys.chatEditingCanRedo, ChatContextKeys.enabled, ChatContextKeys.editingParticipantRegistered),
+				f1: true,
 				menu: [{
-					id: MenuId.ViewTitle,
+					id: MenuId.ChatMessageRestoreCheckpoint,
 					when: ContextKeyExpr.equals('view', ChatViewId),
 					group: 'navigation',
-					order: -2
+					order: -1
 				}]
 			});
 		}
 
 		async runEditingSessionAction(accessor: ServicesAccessor, editingSession: IChatEditingSession) {
-			await editingSession.redoInteraction();
+			const widget = accessor.get(IChatWidgetService);
+
+			const redoUntilDone = async () => {
+				if (!editingSession.canRedo.get()) {
+					return;
+				}
+
+				await editingSession.redoInteraction();
+				setTimeout(() => redoUntilDone(), 0);
```

Comment:
:( hopefully we can find a way to do this without an async loop, which might cause strange issues